feat: Add public OOO notes display on booking pages

dhairyashiil opened the pull request:
### Video:

1.


https://github.com/user-attachments/assets/bc4d96cd-e5b0-4409-9300-87096e312c4f


2. 



https://github.com/user-attachments/assets/b6e260fe-4fed-49b8-be48-a5e5fe032d30


3. tooltip



https://github.com/user-attachments/assets/b7e8ceba-abc7-42e2-b21d-f99333f3f6d4


4. team ooo



https://github.com/user-attachments/assets/647bc590-6198-4234-81d9-7d302987d239




## What does this PR do?

Adds the ability to display Out of Office (OOO) notes publicly on booking pages with a privacy toggle. This feature improves transparency and communication when users are unavailable.

## Changes

### Database Schema
- Added `showNotePublicly` boolean field to `OutOfOfficeEntry` model (default: `false` for backward compatibility)
- Created and applied database migration

### Backend (tRPC)
- Updated `outOfOfficeCreateOrUpdate` handler to accept and save `showNotePublicly` field
- Updated Zod schema validation to include `showNotePublicly` as optional boolean
- Modified OOO repository methods to select and return `notes` and `showNotePublicly` fields
- Updated `getUserAvailability` to propagate OOO notes and visibility through the data flow
- Enhanced slot generation logic to include notes when publicly visible

### Frontend - Settings UI
- Added "Show note publicly" toggle to Create/Edit OOO Entry modal
- Toggle positioned next to "Notes" label following Booking Questions UI pattern
- Shows "Hidden" badge when toggle is OFF, no text when ON
- Toggle always visible (not dependent on note content)
- Fixed bug where toggle state wasn't properly loaded when editing existing entries

### Frontend - Booking Page
- Display OOO message on booking page when `showNotePublicly` is enabled
- Message shows before redirect information with "OOO Message:" label
- Only message text is italicized, not the label
- OOO dates are now selectable even without redirect configured
- Redirect button only shows when team member is assigned
- Added scrollable container for long notes (max-height: 120px) to prevent UI overflow
- Hide 12h/24h time format toggle when viewing OOO dates (optimization: O(1) check)

## Use Cases

### Scenario 1: OOO with Redirect 
- User selects OOO date on calendar 
- Sees OOO emoji, user name, message, and redirect text 
- Redirect button displayed to book with team member 

### Scenario 2: OOO without Redirect 
- User selects OOO date on calendar 
- Sees OOO emoji, user name, and message (if public) 
- No redirect button shown 
- Cannot book time slots (blocked)

dhairyashiil committed:
feat: ooo message on booking page

dhairyashiil committed:
make ooo days selectable even when no redirect booking

dhairyashiil committed:
handle long notes

dhairyashiil committed:
remove unused i18n key

dhairyashiil commented on packages/features/bookings/Booker/components/AvailableTimeSlots.tsx:
```diff
@@ -191,23 +191,27 @@ export const AvailableTimeSlots = ({
-          ))
+          slotsPerDay.map((slots) => {
+            // Check if this day is OOO - since OOO is date-level, just check the first slot
+            const isOOODay = slots.slots.length > 0 && slots.slots[0]?.away;
```
OOO entries are always full-day in the current implementation (see calculateOutOfOfficeRanges which uses .startOf("day") and .endOf("day")). When an OOO entry exists for a date, ALL slots for that day are marked as away: true during slot generation. Therefore, checking only the first slot is sufficient and more performant (O(1) vs O(n)). Partial-day OOO is not currently supported.

dhairyashiil committed:
Private notes stay private on the server, No accidental data leaks through client-side payloads

dhairyashiil committed:
address cubics comments

CarinaWolli reviewed (changes requested):
- Remove "OOO message:" 
<img width="461" height="576" alt="Screenshot 2025-12-05 at 4 01 29 PM" src="https://github.com/user-attachments/assets/6b5613e4-06a7-4db4-96fd-12f11ac7c6a6" />

- Instead of the toggle add a checkbox with "Show note on public booking page"
<img width="472" height="118" alt="Screenshot 2025-12-05 at 4 03 05 PM" src="https://github.com/user-attachments/assets/c0fe3d77-b902-43da-922e-0ddf9fdfe19d" />

- Didn't test, but this also needs to show when there is no redirect

dhairyashiil committed:
Merge remote-tracking branch 'upstream' into feat/ooo-message-on-booking-page

dhairyashiil committed:
fix: replace toggle with checkbox for OOO note visibility

- Replace Switch with Checkbox for "show note publicly" option
- Remove "OOO Message:" prefix from displayed notes on booking page
- Update i18n text to "Show note on public booking page"
- Remove unused ooo_message i18n key

dhairyashiil commented:
done @CarinaWolli , here is the updated video (also shown the case where this work when there is no redirect)


https://github.com/user-attachments/assets/8408a774-6685-4f61-9261-7eb97cc45cf7

dhairyashiil committed:
fix the accessibility issue by using proper htmlFor and id association

dhairyashiil committed:
Merge remote-tracking branch 'upstream/main' into feat/ooo-message-on-booking-page

dhairyashiil committed:
only allow selecting OOO dates when the note is public

CarinaWolli reviewed (approved):
Nice work, looks great 👏🏻

emrysal reviewed (approved):
Approved!
